nutils/function.py

```diff
@@ -65,6 +65,14 @@ def map_coordinates(self, map):
 
         return self if self.coordinates is None else self.replace(coordinates=map(self.coordinates))
 
+    def rename_spaces(self, map: Mapping[str, str]) -> 'LowerArg':
+        '''Return a copy of the :class:`LowerArg` with :attr:`space` renamed using ``map``
+
+        If the old space is not found in ``map``, the :class:`LowerArg` is returned unchanged.
+        '''
+
+        return self if (space := map.get(self.space, self.space)) == self.space else self.replace(space=space)
```

Comment:
I would put the space= assignment on the line above, since there is no reason not to.

Reply:
applied